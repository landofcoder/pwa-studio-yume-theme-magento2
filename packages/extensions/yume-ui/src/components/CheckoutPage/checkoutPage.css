.root {
    padding: 2.5rem 3rem;
    max-width: 1080px;
    margin: 0 auto;
}

.checkoutContent {
    display: grid;
    gap: 2rem;
    grid-template-columns: 2fr 1fr;
}

.checkoutContent_hidden {
    composes: checkoutContent;
    display: none;
}

.heading_container {
    display: grid;
    grid-column-end: span 2;
    row-gap: 1rem;
}

.heading {
    line-height: 1.25em;
}

.cartLink {
    composes: root from '../LinkButton/linkButton.css';
    font-size: inherit;
}

.stepper_heading {
    border-bottom: 1px solid rgb(var(--venia-global-color-border));
    font-size: 1rem;
    font-weight: 600;
    padding-bottom: 2rem;
    text-transform: uppercase;
}

.shipping_method_heading {
    composes: stepper_heading;
}

.payment_information_heading {
    composes: stepper_heading;
}

.signin_container {
    grid-column: 1 / span 1;
    border-bottom: 1px solid rgb(var(--venia-global-color-border));
}

.empty_cart_container {
    margin-bottom: 1.5rem;
}

.shipping_information_container {
    grid-column: 1 / span 1;
}

.shipping_method_container {
    grid-column: 1 / span 1;
}

.payment_information_container {
    grid-column: 1 / span 1;
}

.price_adjustments_container {
    grid-column: 1 / span 1;
}

.items_review_container {
    grid-column: 1 / span 1;
}

.sign_in {
    composes: root from '../LinkButton/linkButton.css';
    margin-bottom: 1.5rem;
}

.summaryContainer {
    grid-column: 1 / span 1;
}

@media (min-width: 961px) {
    .summaryContainer {
        grid-column: 2 / span 1;
        grid-row: 2 / span 3;
        position: sticky;
        top: 5.5rem;
        height: min-content;
    }

    .signin_container ~ .summaryContainer {
        grid-row: 3 / span 3;
    }
}

.review_order_button {
    composes: root_highPriority from '../Button/button.css';
    grid-column: 1 / span 1;
    margin: auto;
}

.place_order_button {
    composes: root_highPriority from '../Button/button.css';
    grid-column: 1 / span 1;
    margin: auto;
}

/*
 * Mobile-specific styles.
 */

@media (max-width: 960px) {
    .root {
        padding-left: 1.5rem;
        padding-right: 1.5rem;
    }

    .checkoutContent {
        /* Only one column in mobile view. */
        grid-template-columns: 1fr;
        gap: 1rem;
    }

    .stepper_heading {
        padding-bottom: 1rem;
    }

    .payment_information_heading {
        composes: stepper_heading;
        /* On mobile, order summary has a top border, so avoid doubling up. */
        border-bottom-width: 0;
    }
}
