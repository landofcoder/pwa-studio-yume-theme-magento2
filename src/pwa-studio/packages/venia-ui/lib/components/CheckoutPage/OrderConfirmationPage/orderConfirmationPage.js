import React, { useEffect } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { object, shape, string } from 'prop-types';
import { useOrderConfirmationPage } from '@magento/peregrine/lib/talons/CheckoutPage/OrderConfirmationPage/useOrderConfirmationPage';

import { mergeClasses } from '../../../classify';
import { Title } from '../../../components/Head';
import CreateAccount from './createAccount';
import ItemsReview from '../ItemsReview';
import defaultClasses from './orderConfirmationPage.css';

const OrderConfirmationPage = props => {
    const classes = mergeClasses(defaultClasses, props.classes);
    const { data, orderNumber } = props;
    const { formatMessage } = useIntl();

    const talonProps = useOrderConfirmationPage({
        data
    });

    const { flatData, isSignedIn } = talonProps;

    const {
        city,
        country,
        email,
        firstname,
        lastname,
        postcode,
        region,
        shippingMethod,
        street
    } = flatData;

    const streetRows = street.map((row, index) => {
        return (
            <span key={index} className={classes.addressStreet}>
                {row}
            </span>
        );
    });

    useEffect(() => {
        window.scrollTo({
            left: 0,
            top: 0,
            behavior: 'smooth'
        });
    }, []);

    const createAccountForm = !isSignedIn ? (
        <CreateAccount
            firstname={firstname}
            lastname={lastname}
            email={email}
        />
    ) : null;

    const nameString = `${firstname} ${lastname}`;
    const additionalAddressString = `${city}, ${region} ${postcode} ${country}`;

    return (
        <div className={classes.root}>
            <Title>
                {formatMessage(
                    {
                        id: 'checkoutPage.titleReceipt',
                        defaultMessage: 'Receipt'
                    },
                    { name: STORE_NAME }
                )}
            </Title>
            <div className={classes.mainContainer}>
                <h2 className={classes.heading}>
                    <FormattedMessage
                        id={'checkoutPage.thankYou'}
                        defaultMessage={'Thank you for your order!'}
                    />
                </h2>
                <div className={classes.orderNumber}>
                    <FormattedMessage
                        id={'checkoutPage.orderNumber'}
                        defaultMessage={'Order Number'}
                        values={{ orderNumber }}
                    />
                </div>
                <div className={classes.shippingInfoHeading}>
                    <FormattedMessage
                        id={'global.shippingInformation'}
                        defaultMessage={'Shipping Information'}
                    />
                </div>
                <div className={classes.shippingInfo}>
                    <span className={classes.email}>{email}</span>
                    <span className={classes.name}>{nameString}</span>
                    {streetRows}
                    <span className={classes.addressAdditional}>
                        {additionalAddressString}
                    </span>
                </div>
                <div className={classes.shippingMethodHeading}>
                    <FormattedMessage
                        id={'global.shippingMethod'}
                        defaultMessage={'Shipping Method'}
                    />
                </div>
                <div className={classes.shippingMethod}>{shippingMethod}</div>
                <div className={classes.itemsReview}>
                    <ItemsReview data={data} />
                </div>
                <div className={classes.additionalText}>
                    <FormattedMessage
                        id={'checkoutPage.additionalText'}
                        defaultMessage={
                            'You will also receive an email with the details and we will let you know when your order has shipped.'
                        }
                    />
                </div>
            </div>
            <div className={classes.sidebarContainer}>{createAccountForm}</div>
        </div>
    );
};

export default OrderConfirmationPage;

OrderConfirmationPage.propTypes = {
    classes: shape({
        addressStreet: string,
        mainContainer: string,
        heading: string,
        orderNumber: string,
        shippingInfoHeading: string,
        shippingInfo: string,
        email: string,
        name: string,
        addressAdditional: string,
        shippingMethodHeading: string,
        shippingMethod: string,
        itemsReview: string,
        additionalText: string,
        sidebarContainer: string
    }),
    data: object.isRequired,
    orderNumber: string
};
