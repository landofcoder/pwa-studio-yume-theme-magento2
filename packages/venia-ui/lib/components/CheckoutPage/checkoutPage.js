import React, { Fragment, useEffect } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { AlertCircle as AlertCircleIcon } from 'react-feather';
import { Link } from 'react-router-dom';

import { useWindowSize, useToasts } from '@magento/peregrine';
import {
    CHECKOUT_STEP,
    useCheckoutPage
} from '@magento/peregrine/lib/talons/CheckoutPage/useCheckoutPage';

import { mergeClasses } from '../../classify';
import Button from '../Button';
import { Title } from '../Head';
import Icon from '../Icon';
import LinkButton from '../LinkButton';
import { fullPageLoadingIndicator } from '../LoadingIndicator';
import StockStatusMessage from '../StockStatusMessage';
import AddressBook from './AddressBook';
import OrderSummary from './OrderSummary';
import PaymentInformation from './PaymentInformation';
import PriceAdjustments from './PriceAdjustments';
import ShippingMethod from './ShippingMethod';
import ShippingInformation from './ShippingInformation';
import OrderConfirmationPage from './OrderConfirmationPage';
import ItemsReview from './ItemsReview';
import defaultClasses from './checkoutPage.css';
import CheckoutPageOperations from './checkoutPage.gql.js';

const errorIcon = <Icon src={AlertCircleIcon} size={20} />;

const CheckoutPage = props => {
    const { classes: propClasses } = props;
    const { formatMessage } = useIntl();
    const talonProps = useCheckoutPage({
        ...CheckoutPageOperations
    });

    const {
        /**
         * Enum, one of:
         * SHIPPING_ADDRESS, SHIPPING_METHOD, PAYMENT, REVIEW
         */
        activeContent,
        cartItems,
        checkoutStep,
        customer,
        error,
        handleSignIn,
        handlePlaceOrder,
        hasError,
        isCartEmpty,
        isGuestCheckout,
        isLoading,
        isUpdating,
        orderDetailsData,
        orderDetailsLoading,
        orderNumber,
        placeOrderLoading,
        setCheckoutStep,
        setIsUpdating,
        setShippingInformationDone,
        setShippingMethodDone,
        setPaymentInformationDone,
        resetReviewOrderButtonClicked,
        handleReviewOrder,
        reviewOrderButtonClicked,
        toggleActiveContent
    } = talonProps;

    const [, { addToast }] = useToasts();

    useEffect(() => {
        if (hasError) {
            const message =
                error && error.message
                    ? error.message
                    : formatMessage({
                          id: 'checkoutPage.errorSubmit',
                          defaultMessage:
                              'Oops! An error occurred while submitting. Please try again.'
                      });
            addToast({
                type: 'error',
                icon: errorIcon,
                message,
                dismissable: true,
                timeout: 7000
            });

            if (process.env.NODE_ENV !== 'production') {
                console.error(error);
            }
        }
    }, [addToast, error, formatMessage, hasError]);

    const classes = mergeClasses(defaultClasses, propClasses);

    const windowSize = useWindowSize();
    const isMobile = windowSize.innerWidth <= 960;

    let checkoutContent;

    const heading = isGuestCheckout
        ? formatMessage({
              id: 'checkoutPage.guestCheckout',
              defaultMessage: 'Guest Checkout'
          })
        : formatMessage({
              id: 'checkoutPage.checkout',
              defaultMessage: 'Checkout'
          });

    if (orderNumber) {
        return (
            <OrderConfirmationPage
                data={orderDetailsData}
                orderNumber={orderNumber}
            />
        );
    } else if (isLoading) {
        return fullPageLoadingIndicator;
    } else if (isCartEmpty) {
        checkoutContent = (
            <div className={classes.empty_cart_container}>
                <div className={classes.heading_container}>
                    <h1 className={classes.heading}>{heading}</h1>
                </div>
                <h3>
                    <FormattedMessage
                        id={'checkoutPage.emptyMessage'}
                        defaultMessage={'There are no items in your cart.'}
                    />
                </h3>
            </div>
        );
    } else {
        const loginButton = isGuestCheckout ? (
            <div className={classes.signin_container}>
                <LinkButton className={classes.sign_in} onClick={handleSignIn}>
                    <FormattedMessage
                        id={'checkoutPage.loginAndCheckoutFaster'}
                        defaultMessage={'Login and Checkout Faster'}
                    />
                </LinkButton>
            </div>
        ) : null;

        const shippingMethodSection =
            checkoutStep >= CHECKOUT_STEP.SHIPPING_METHOD ? (
                <ShippingMethod
                    pageIsUpdating={isUpdating}
                    onSave={setShippingMethodDone}
                    setPageIsUpdating={setIsUpdating}
                />
            ) : (
                <h3 className={classes.shipping_method_heading}>
                    <FormattedMessage
                        id={'checkoutPage.shippingMethodStep'}
                        defaultMessage={'2. Shipping Method'}
                    />
                </h3>
            );

        const paymentInformationSection =
            checkoutStep >= CHECKOUT_STEP.PAYMENT ? (
                <PaymentInformation
                    onSave={setPaymentInformationDone}
                    checkoutError={error}
                    resetShouldSubmit={resetReviewOrderButtonClicked}
                    setCheckoutStep={setCheckoutStep}
                    shouldSubmit={reviewOrderButtonClicked}
                />
            ) : (
                <h3 className={classes.payment_information_heading}>
                    <FormattedMessage
                        id={'checkoutPage.paymentInformationStep'}
                        defaultMessage={'3. Payment Information'}
                    />
                </h3>
            );

        const priceAdjustmentsSection =
            checkoutStep === CHECKOUT_STEP.PAYMENT ? (
                <div className={classes.price_adjustments_container}>
                    <PriceAdjustments setPageIsUpdating={setIsUpdating} />
                </div>
            ) : null;

        const reviewOrderButton =
            checkoutStep === CHECKOUT_STEP.PAYMENT ? (
                <Button
                    onClick={handleReviewOrder}
                    priority="high"
                    className={classes.review_order_button}
                    disabled={reviewOrderButtonClicked || isUpdating}
                >
                    <FormattedMessage
                        id={'checkoutPage.reviewOrder'}
                        defaultMessage={'Review Order'}
                    />
                </Button>
            ) : null;

        const itemsReview =
            checkoutStep === CHECKOUT_STEP.REVIEW ? (
                <div className={classes.items_review_container}>
                    <ItemsReview />
                </div>
            ) : null;

        const placeOrderButton =
            checkoutStep === CHECKOUT_STEP.REVIEW ? (
                <Button
                    onClick={handlePlaceOrder}
                    priority="high"
                    className={classes.place_order_button}
                    disabled={
                        isUpdating || placeOrderLoading || orderDetailsLoading
                    }
                >
                    <FormattedMessage
                        id={'checkoutPage.placeOrder'}
                        defaultMessage={'Place Order'}
                    />
                </Button>
            ) : null;

        // If we're on mobile we should only render price summary in/after review.
        const shouldRenderPriceSummary = !(
            isMobile && checkoutStep < CHECKOUT_STEP.REVIEW
        );

        const orderSummary = shouldRenderPriceSummary ? (
            <div className={classes.summaryContainer}>
                <OrderSummary isUpdating={isUpdating} />
            </div>
        ) : null;

        let headerText;

        if (isGuestCheckout) {
            headerText = formatMessage({
                id: 'checkoutPage.guestCheckout',
                defaultMessage: 'Guest Checkout'
            });
        } else if (customer.default_shipping) {
            headerText = formatMessage({
                id: 'checkoutPage.reviewAndPlaceOrder',
                defaultMessage: 'Review and Place Order'
            });
        } else {
            headerText = formatMessage(
                { id: 'checkoutPage.greeting', defaultMessage: 'Welcome' },
                { firstname: customer.firstname }
            );
        }

        const checkoutContentClass =
            activeContent === 'checkout'
                ? classes.checkoutContent
                : classes.checkoutContent_hidden;

        const stockStatusMessageElement = (
            <Fragment>
                <FormattedMessage
                    id={'checkoutPage.stockStatusMessage'}
                    defaultMessage={
                        'An item in your cart is currently out-of-stock and must be removed in order to Checkout. Please return to your cart to remove the item.'
                    }
                />
                <Link className={classes.cartLink} to={'/cart'}>
                    <FormattedMessage
                        id={'checkoutPage.returnToCart'}
                        defaultMessage={'Return to Cart'}
                    />
                </Link>
            </Fragment>
        );
        checkoutContent = (
            <div className={checkoutContentClass}>
                {loginButton}
                <div className={classes.heading_container}>
                    <StockStatusMessage
                        cartItems={cartItems}
                        message={stockStatusMessageElement}
                    />
                    <h1 className={classes.heading}>{headerText}</h1>
                </div>
                <div className={classes.shipping_information_container}>
                    <ShippingInformation
                        onSave={setShippingInformationDone}
                        toggleActiveContent={toggleActiveContent}
                    />
                </div>
                <div className={classes.shipping_method_container}>
                    {shippingMethodSection}
                </div>
                <div className={classes.payment_information_container}>
                    {paymentInformationSection}
                </div>
                {priceAdjustmentsSection}
                {reviewOrderButton}
                {itemsReview}
                {orderSummary}
                {placeOrderButton}
            </div>
        );
    }

    const addressBookElement = !isGuestCheckout ? (
        <AddressBook
            activeContent={activeContent}
            toggleActiveContent={toggleActiveContent}
        />
    ) : null;

    return (
        <div className={classes.root}>
            <Title>
                {formatMessage(
                    {
                        id: 'checkoutPage.titleCheckout',
                        defaultMessage: 'Checkout'
                    },
                    { name: STORE_NAME }
                )}
            </Title>
            {checkoutContent}
            {addressBookElement}
        </div>
    );
};

export default CheckoutPage;
