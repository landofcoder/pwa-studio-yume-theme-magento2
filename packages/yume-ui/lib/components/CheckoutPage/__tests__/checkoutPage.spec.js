import React from 'react';
import { createTestInstance, useToasts } from '@magento/peregrine';

import {
    CHECKOUT_STEP,
    useCheckoutPage
} from '@magento/peregrine/lib/talons/CheckoutPage/useCheckoutPage';
import CheckoutPage from '../checkoutPage';
import OrderConfirmationPage from '../OrderConfirmationPage';

jest.mock('@magento/peregrine', () => {
    const actual = jest.requireActual('@magento/peregrine');
    const useToasts = jest.fn().mockReturnValue([{}, { addToast: jest.fn() }]);
    const useWindowSize = jest.fn().mockReturnValue({
        innerWidth: 961
    });

    return {
        createTestInstance: actual.createTestInstance,
        useToasts,
        useWindowSize
    };
});

jest.mock('@magento/peregrine/lib/talons/CheckoutPage/useCheckoutPage', () => {
    const originalModule = jest.requireActual(
        '@magento/peregrine/lib/talons/CheckoutPage/useCheckoutPage'
    );

    return {
        CHECKOUT_STEP: originalModule.CHECKOUT_STEP,
        useCheckoutPage: jest.fn()
    };
});

jest.mock('../../../classify');

jest.mock('../../../components/Head', () => ({ Title: () => 'Title' }));
jest.mock('../../StockStatusMessage', () => 'StockStatusMessage');
jest.mock('../ItemsReview', () => 'ItemsReview');
jest.mock('../OrderSummary', () => 'OrderSummary');
jest.mock('../OrderConfirmationPage', () => 'OrderConfirmationPage');
jest.mock('../ShippingInformation', () => 'ShippingInformation');
jest.mock('../ShippingMethod', () => 'ShippingMethod');
jest.mock('../PaymentInformation', () => 'PaymentInformation');
jest.mock('../PriceAdjustments', () => 'PriceAdjustments');
jest.mock('../AddressBook', () => 'AddressBook');

const defaultTalonProps = {
    activeContent: 'checkout',
    cartItems: [],
    checkoutStep: 1,
    customer: null,
    error: false,
    handleSignIn: jest.fn().mockName('handleSignIn'),
    handlePlaceOrder: jest.fn().mockName('handlePlaceOrder'),
    hasError: false,
    isCartEmpty: false,
    isGuestCheckout: true,
    isLoading: false,
    isUpdating: false,
    orderDetailsData: null,
    orderDetailsLoading: false,
    orderNumber: null,
    placeOrderLoading: false,
    setIsUpdating: jest.fn().mockName('setIsUpdating'),
    setShippingInformationDone: jest
        .fn()
        .mockName('setShippingInformationDone'),
    setShippingMethodDone: jest.fn().mockName('setShippingMethodDone'),
    setPaymentInformationDone: jest.fn().mockName('setPaymentInformationDone'),
    toggleActiveContent: jest.fn().mockName('toggleActiveContent')
};
describe('CheckoutPage', () => {
    test('throws a toast if there is an error', () => {
        useCheckoutPage.mockReturnValueOnce({
            ...defaultTalonProps,
            hasError: true
        });
        const [, { addToast }] = useToasts();

        createTestInstance(<CheckoutPage />);

        expect(addToast).toHaveBeenCalled();
    });

    test('renders order confirmation page', () => {
        useCheckoutPage.mockReturnValueOnce({
            ...defaultTalonProps,
            placeOrderLoading: false,
            hasError: false,
            orderDetailsData: {},
            orderNumber: 1
        });

        const instance = createTestInstance(<CheckoutPage />);
        const component = instance.root.findByType(OrderConfirmationPage);
        expect(component).toBeTruthy();
    });

    test('disables place order button while loading', () => {
        useCheckoutPage.mockReturnValueOnce({
            ...defaultTalonProps,
            checkoutStep: CHECKOUT_STEP.REVIEW,
            isUpdating: true,
            placeOrderLoading: true,
            orderDetailsLoading: true,
            orderNumber: null
        });

        const instance = createTestInstance(<CheckoutPage />);
        const button = instance.root.findByProps({
            className: 'place_order_button'
        });

        expect(button).toBeTruthy();
        expect(button.props.disabled).toBe(true);
    });

    test('renders loading indicator', () => {
        useCheckoutPage.mockReturnValueOnce({
            isLoading: true
        });

        const tree = createTestInstance(<CheckoutPage />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('renders checkout content for guest', () => {
        useCheckoutPage.mockReturnValueOnce(defaultTalonProps);

        const tree = createTestInstance(<CheckoutPage />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('renders checkout content for customer - no default address', () => {
        useCheckoutPage.mockReturnValueOnce({
            ...defaultTalonProps,
            customer: { default_shipping: null, firstname: 'Eloise' },
            isGuestCheckout: false
        });

        const tree = createTestInstance(<CheckoutPage />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('renders checkout content for customer - default address', () => {
        useCheckoutPage.mockReturnValueOnce({
            ...defaultTalonProps,
            customer: { default_shipping: '1' },
            isGuestCheckout: false
        });

        const tree = createTestInstance(<CheckoutPage />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('renders address book for customer', () => {
        useCheckoutPage.mockReturnValueOnce({
            ...defaultTalonProps,
            activeContent: 'addressBook',
            customer: { default_shipping: '1' },
            isGuestCheckout: false
        });

        const tree = createTestInstance(<CheckoutPage />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
