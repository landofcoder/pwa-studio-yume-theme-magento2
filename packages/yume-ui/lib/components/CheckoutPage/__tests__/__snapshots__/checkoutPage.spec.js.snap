// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage renders address book for customer 1`] = `
<div
  className="root"
>
  Title
  <div
    className="checkoutContent_hidden"
  >
    <div
      className="heading_container"
    >
      <StockStatusMessage
        cartItems={Array []}
        message={
          <React.Fragment>
            <mockConstructor
              defaultMessage="An item in your cart is currently out-of-stock and must be removed in order to Checkout. Please return to your cart to remove the item."
              id="checkoutPage.stockStatusMessage"
            />
            <Link
              className="cartLink"
              to="/cart"
            >
              <mockConstructor
                defaultMessage="Return to Cart"
                id="checkoutPage.returnToCart"
              />
            </Link>
          </React.Fragment>
        }
      />
      <h1
        className="heading"
      >
        Review and Place Order
      </h1>
    </div>
    <div
      className="shipping_information_container"
    >
      <ShippingInformation
        onSave={[MockFunction setShippingInformationDone]}
        toggleActiveContent={[MockFunction toggleActiveContent]}
      />
    </div>
    <div
      className="shipping_method_container"
    >
      <h3
        className="shipping_method_heading"
      >
        2. Shipping Method
      </h3>
    </div>
    <div
      className="payment_information_container"
    >
      <h3
        className="payment_information_heading"
      >
        3. Payment Information
      </h3>
    </div>
    <div
      className="summaryContainer"
    >
      <OrderSummary
        isUpdating={false}
      />
    </div>
  </div>
  <AddressBook
    activeContent="addressBook"
    toggleActiveContent={[MockFunction toggleActiveContent]}
  />
</div>
`;

exports[`CheckoutPage renders checkout content for customer - default address 1`] = `
<div
  className="root"
>
  Title
  <div
    className="checkoutContent"
  >
    <div
      className="heading_container"
    >
      <StockStatusMessage
        cartItems={Array []}
        message={
          <React.Fragment>
            <mockConstructor
              defaultMessage="An item in your cart is currently out-of-stock and must be removed in order to Checkout. Please return to your cart to remove the item."
              id="checkoutPage.stockStatusMessage"
            />
            <Link
              className="cartLink"
              to="/cart"
            >
              <mockConstructor
                defaultMessage="Return to Cart"
                id="checkoutPage.returnToCart"
              />
            </Link>
          </React.Fragment>
        }
      />
      <h1
        className="heading"
      >
        Review and Place Order
      </h1>
    </div>
    <div
      className="shipping_information_container"
    >
      <ShippingInformation
        onSave={[MockFunction setShippingInformationDone]}
        toggleActiveContent={[MockFunction toggleActiveContent]}
      />
    </div>
    <div
      className="shipping_method_container"
    >
      <h3
        className="shipping_method_heading"
      >
        2. Shipping Method
      </h3>
    </div>
    <div
      className="payment_information_container"
    >
      <h3
        className="payment_information_heading"
      >
        3. Payment Information
      </h3>
    </div>
    <div
      className="summaryContainer"
    >
      <OrderSummary
        isUpdating={false}
      />
    </div>
  </div>
  <AddressBook
    activeContent="checkout"
    toggleActiveContent={[MockFunction toggleActiveContent]}
  />
</div>
`;

exports[`CheckoutPage renders checkout content for customer - no default address 1`] = `
<div
  className="root"
>
  Title
  <div
    className="checkoutContent"
  >
    <div
      className="heading_container"
    >
      <StockStatusMessage
        cartItems={Array []}
        message={
          <React.Fragment>
            <mockConstructor
              defaultMessage="An item in your cart is currently out-of-stock and must be removed in order to Checkout. Please return to your cart to remove the item."
              id="checkoutPage.stockStatusMessage"
            />
            <Link
              className="cartLink"
              to="/cart"
            >
              <mockConstructor
                defaultMessage="Return to Cart"
                id="checkoutPage.returnToCart"
              />
            </Link>
          </React.Fragment>
        }
      />
      <h1
        className="heading"
      >
        Welcome Eloise!
      </h1>
    </div>
    <div
      className="shipping_information_container"
    >
      <ShippingInformation
        onSave={[MockFunction setShippingInformationDone]}
        toggleActiveContent={[MockFunction toggleActiveContent]}
      />
    </div>
    <div
      className="shipping_method_container"
    >
      <h3
        className="shipping_method_heading"
      >
        2. Shipping Method
      </h3>
    </div>
    <div
      className="payment_information_container"
    >
      <h3
        className="payment_information_heading"
      >
        3. Payment Information
      </h3>
    </div>
    <div
      className="summaryContainer"
    >
      <OrderSummary
        isUpdating={false}
      />
    </div>
  </div>
  <AddressBook
    activeContent="checkout"
    toggleActiveContent={[MockFunction toggleActiveContent]}
  />
</div>
`;

exports[`CheckoutPage renders checkout content for guest 1`] = `
<div
  className="root"
>
  Title
  <div
    className="checkoutContent"
  >
    <div
      className="signin_container"
    >
      <button
        className="sign_in"
        disabled={false}
        onClick={[MockFunction handleSignIn]}
        type="button"
      >
        <span
          className="content"
        >
          Login and Checkout Faster
        </span>
      </button>
    </div>
    <div
      className="heading_container"
    >
      <StockStatusMessage
        cartItems={Array []}
        message={
          <React.Fragment>
            <mockConstructor
              defaultMessage="An item in your cart is currently out-of-stock and must be removed in order to Checkout. Please return to your cart to remove the item."
              id="checkoutPage.stockStatusMessage"
            />
            <Link
              className="cartLink"
              to="/cart"
            >
              <mockConstructor
                defaultMessage="Return to Cart"
                id="checkoutPage.returnToCart"
              />
            </Link>
          </React.Fragment>
        }
      />
      <h1
        className="heading"
      >
        Guest Checkout
      </h1>
    </div>
    <div
      className="shipping_information_container"
    >
      <ShippingInformation
        onSave={[MockFunction setShippingInformationDone]}
        toggleActiveContent={[MockFunction toggleActiveContent]}
      />
    </div>
    <div
      className="shipping_method_container"
    >
      <h3
        className="shipping_method_heading"
      >
        2. Shipping Method
      </h3>
    </div>
    <div
      className="payment_information_container"
    >
      <h3
        className="payment_information_heading"
      >
        3. Payment Information
      </h3>
    </div>
    <div
      className="summaryContainer"
    >
      <OrderSummary
        isUpdating={false}
      />
    </div>
  </div>
</div>
`;

exports[`CheckoutPage renders loading indicator 1`] = `
<div
  className="global"
>
  <span
    className="root"
  >
    <svg
      className="icon"
      fill="none"
      height={64}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={64}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="12"
        x2="12"
        y1="2"
        y2="6"
      />
      <line
        x1="12"
        x2="12"
        y1="18"
        y2="22"
      />
      <line
        x1="4.93"
        x2="7.76"
        y1="4.93"
        y2="7.76"
      />
      <line
        x1="16.24"
        x2="19.07"
        y1="16.24"
        y2="19.07"
      />
      <line
        x1="2"
        x2="6"
        y1="12"
        y2="12"
      />
      <line
        x1="18"
        x2="22"
        y1="12"
        y2="12"
      />
      <line
        x1="4.93"
        x2="7.76"
        y1="19.07"
        y2="16.24"
      />
      <line
        x1="16.24"
        x2="19.07"
        y1="7.76"
        y2="4.93"
      />
    </svg>
  </span>
  <span
    className="message"
  >
    Fetching Data...
  </span>
</div>
`;
